test/tests/functional/pbs_suspend_resume_accounting.py

```diff
@@ -0,0 +1,356 @@
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+
+
+class TestSuspendResumeAccounting(TestFunctional):
+    """
+    Testsuite for verifying accounting of
+    suspend/resume events of job
+    """
+
+    script = ['#!/bin/bash\nfor ((c=1; c <= 1000000000; c++));']
+    script += ['do']
+    script += ['sleep 1']
+    script += ['done']
+
+    def test_suspend_resume_job_signal(self):
+        """
+        Test case to verify accounting suspend
+        and resume records when the events are
+        triggered by client for one job.
+        """
+        j = Job(TEST_USER)
+        j.create_script(self.script)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+
+        self.server.sigjob(jobid=jid, signal="suspend")
+
+        record = 'z;%s;requestor=%s@%s .*resources_used.' % \
+                 (jid, self.server.current_user, self.server.hostname)
+        self.server.accounting_match(msg=record, id=jid, regexp=True)
+
+        self.server.sigjob(jobid=jid, signal="resume")
+        record = 'r;%s;requestor=Scheduler@%s' % \
+                 (jid, self.server.hostname)
+        self.server.accounting_match(msg=record, id=jid)
+
+    @skipOnCpuSet
+    def test_suspend_resume_job_array_signal(self):
+        """
+        Test case to verify accounting suspend
+        and resume records when the events are
+        triggered by client for a job array.
+        """
+        a = {ATTR_rescavail + '.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        j = Job(TEST_USER)
+        j.create_script(self.script)
+        j.set_attributes({ATTR_J: '1-2'})
+        jid = self.server.submit(j)
+
+        subjobs = self.server.status(JOB, id=jid, extend='t')
+        sub_jid1 = subjobs[1]['id']
+        sub_jid2 = subjobs[2]['id']
```

Comment:
You can use create_subjob_id() instead if you want.  It'll save the status() call.